### PR TITLE
[Enhancement] Support only deleting shard meta from fe for shared-data cluster

### LINKED ISSUE #53554: Support fast deletion of shard meta from FE in shared-data
## Enhancement

Fast delete large volume of shard meta from FE to lower GC overhead in shared-data mode cluster.

### PR DESCRIPTION
**Pull request by wxl24life**
## Why I'm doing:

Add an option to delete staros meta from FE quickly

## What I'm doing:

Fixes #53554 

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.4
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0

### CODE REVIEW

**Review by starrocks-xupeng — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/lake/StarMgrMetaSyncer.java [hunk 1]
```diff
@@ -217,6 +205,44 @@ private void deleteUnusedShardAndShardGroup() {
...
+            // delete shard from star manager only, not considering tablet data on be/cn
+            if (Config.meta_sync_force_delete_shard_meta) {
+                LOG.debug("delete shards from starMgr only, shard group: {}", groupId);
+                try {
+                    // all shards have the same root directory
+                    long shardId = shardIds.get(0);
+                    ShardInfo shardInfo = starOSAgent.getShardInfo(shardId, StarOSAgent.DEFAULT_WORKER_GROUP_ID);
+                    // record the root directory of this shard group, with format like  s3://bucket/xx/db15570/15648/15944
+                    String rootDirectory = shardInfo.getFilePath().getFullPath();
+                    starOSAgent.deleteShards(new HashSet<>(shardIds));
```
Comment (starrocks-xupeng) [inline fe/fe-core/src/main/java/com/starrocks/lake/StarMgrMetaSyncer.java:224]: catch getShardInfo exception, if getShardInfo fails, still call deleteShards

File: fe/fe-core/src/test/java/com/starrocks/lake/StarMgrMetaSyncerTest.java [hunk 2]
```diff
@@ -733,4 +827,9 @@ public ComputeNode getBackendOrComputeNode(long nodeId) {
         starMgrMetaSyncer.syncTableMetaInternal(db, (OlapTable) table, true);
         Assert.assertEquals(3, shards.size());
     }
+
+    @Test
+    public void testGetAllPartitionShardGroupIdToTableInvertedIndex() {
```
Comment (starrocks-xupeng) [inline fe/fe-core/src/test/java/com/starrocks/lake/StarMgrMetaSyncerTest.java:832]: remoge this

Comment (kevincai) [inline fe/fe-core/src/test/java/com/starrocks/lake/StarMgrMetaSyncerTest.java:832]: an empty UT?

Comment (wxl24life) [reply fe/fe-core/src/test/java/com/starrocks/lake/StarMgrMetaSyncerTest.java:832]: oops

Comment (wxl24life) [reply fe/fe-core/src/test/java/com/starrocks/lake/StarMgrMetaSyncerTest.java]: done

File: fe/fe-core/src/main/java/com/starrocks/lake/StarMgrMetaSyncer.java [hunk 3]
```diff
@@ -217,6 +205,58 @@ private void deleteUnusedShardAndShardGroup() {
...
+            LOG.warn("failed to get shard root directory from starMgr, shard id {}, {}", shardId,
+                    e.getMessage());
+        }
+        try {
+            starOSAgent.deleteShards(new HashSet<>(shardIds));
+            if (StringUtils.isNotEmpty(rootDirectory)) {
+                LOG.info(
+                        "shard group {} deleted from starMgr only, you may need to delete remote file path manually," +
+                                " file path is: {}", groupId, rootDirectory);
+            } else {
```
Comment (starrocks-xupeng) [inline fe/fe-core/src/main/java/com/starrocks/lake/StarMgrMetaSyncer.java:252]: remove this else, log is already printed in line 243

Comment (wxl24life) [reply fe/fe-core/src/main/java/com/starrocks/lake/StarMgrMetaSyncer.java]: done

**Review by kevincai — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Pull request merged**